frontend/src/features/notifications/components/NotificationInformationModal.tsx

```diff
@@ -44,158 +44,235 @@ const isValidDemandCapacityNotification = (notification: Partial<DemandCapacityN
-                {demandCapacityNotification.affectedMaterialNumbers && demandCapacityNotification.affectedMaterialNumbers.length > 0
-                    ? demandCapacityNotification.affectedMaterialNumbers.join(', ')
-                    : 'None'}
-            </Grid>
-        </Grid>
-    );
-};
-
 export const DemandCapacityNotificationInformationModal = ({
     open,
+    mode,
     demandCapacityNotification,
     onClose,
     onSave,
 }: DemandCapacityNotificationInformationModalProps) => {
-    const [temporaryDemandCapacityNotification, setTemporaryDemandCapacityNotification] = useState<Partial<DemandCapacityNotification>>({});
+    const [temporaryDemandCapacityNotification, setTemporaryDemandCapacityNotification] = useState<Partial<DemandCapacityNotification>>(demandCapacityNotification ?? {});
     const { partners } = useAllPartners();
-    const { partnerMaterials } = usePartnerMaterials(temporaryDemandCapacityNotification.partnerBpnl ?? `BPNL`);
-
+    const [availablePartners, setAvailablePartners] = useState(partners);
+    const { partnerMaterials } = usePartnerMaterials(temporaryDemandCapacityNotification.partnerBpnl);
     const [notifications, setNotifications] = useState<Notification[]>([]);
     const [formError, setFormError] = useState(false);
-
     const { sites } = useSites();
 
     useEffect(() => {
-        setTemporaryDemandCapacityNotification((prevState) => ({
-            ...prevState,
+        if (mode === 'react') {
+            resetAndUpdateNotification();
+        }
+        else {
+            updateNotification();
+        }
+        updateAvailablePartners();
+    }, [demandCapacityNotification, mode, partners]);
+
+    useEffect(() => {
+        resetAffectedFields();
+    }, [temporaryDemandCapacityNotification.partnerBpnl]);
+
+    const resetAndUpdateNotification = () => {
+        setTemporaryDemandCapacityNotification({
+            ...demandCapacityNotification,
+            partnerBpnl: undefined,
             affectedMaterialNumbers: [],
             affectedSitesBpnsRecipient: [],
-        }));
-    }, [temporaryDemandCapacityNotification.partnerBpnl]);
+            affectedSitesBpnsSender: [],
+        });
+    };
+
+    const updateNotification = () => {
+        setTemporaryDemandCapacityNotification(demandCapacityNotification ?? {});
+    };
+
+    const updateAvailablePartners = () => {
+        if (partners !== null) {
+            const newFilteredPartners = partners.filter((partner) => partner.bpnl !== demandCapacityNotification?.partnerBpnl);
+            setAvailablePartners(newFilteredPartners);
+        }
+    };
+
+    const resetAffectedFields = () => {
+        if (temporaryDemandCapacityNotification.partnerBpnl) {
+            setTemporaryDemandCapacityNotification((prevState) => ({
+                ...prevState,
+                affectedMaterialNumbers: [],
+                affectedSitesBpnsRecipient: [],
+            }));
+        }
+    };
```

Comment:
Move these 4 functions into their respective useEffect. For the first one it might be better to extract into a custom hook like useTemporaryDemandAndCapacityNotification.